src/Parser/Priest/Shared/Modules/Features/DesperatePrayer.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import ExpandableStatisticBox from 'Main/ExpandableStatisticBox';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class DesperatePrayer extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  desperatePrayerUsages = [];
+  currentDesperatePrayerUsage;
+  deathsWithDPReady = 0;
+
+  on_byPlayer_applybuff(event) {
+    if (event.ability.guid !== SPELLS.DESPERATE_PRAYER.id) {
+      return;
+    }
+    this.currentDesperatePrayerUsage = {
+      DamageTaken: 0,
+      HPWhenUsed: 0,
+      MaxHPWhenUsed: 0,
+    };
+  }
+
+  on_toPlayer_heal(event) {
+    if (event.ability.guid !== SPELLS.DESPERATE_PRAYER.id) {
+      return;
+    }
+    this.currentDesperatePrayerUsage.HPWhenUsed = event.hitPoints - event.amount;
+    this.currentDesperatePrayerUsage.MaxHPWhenUsed = event.maxHitPoints;
+  }
+
+  on_toPlayer_damage(event) {
+    if(!this.combatants.selected.hasBuff(SPELLS.DESPERATE_PRAYER.id)) {
+      return;
+    }
+    this.currentDesperatePrayerUsage.DamageTaken += event.amount;
+  }
+
+  on_byPlayer_removebuff(event) {
+    if (event.ability.guid !== SPELLS.DESPERATE_PRAYER.id) {
+      return;
+    }
+    this.desperatePrayerUsages.push(this.currentDesperatePrayerUsage);
```

Comment:
Activations at the end of the fight will be missed with this. You might not even need the `currentDesperatePrayerUsage` property and could instead refer to the last item in the array.